e2e/connect/connect_test.go

```diff
@@ -120,6 +121,30 @@ func testConnectMultiService(t *testing.T) {
 	assertServiceOk(t, cc, "echo2-sidecar-proxy")
 }
 
+// testConnectTransparentProxy tests the Connect Transparent Proxy integration
+func testConnectTransparentProxy(t *testing.T) {
+	_, cleanup := jobs3.Submit(t, "./input/tproxy.nomad.hcl", jobs3.Timeout(time.Second*60))
+	t.Cleanup(cleanup)
+
+	cc := e2eutil.ConsulClient(t)
+
+	ixn := &capi.Intention{
+		SourceName:      "count-dashboard",
+		DestinationName: "count-api",
+		Action:          "allow",
+	}
+	_, err := cc.Connect().IntentionUpsert(ixn, nil)
+	must.NoError(t, err, must.Sprint("could not create intention"))
+
+	t.Cleanup(func() {
+		_, err := cc.Connect().IntentionDeleteExact("count-dashboard", "count-api", nil)
+		test.NoError(t, err)
+	})
+
+	assertServiceOk(t, cc, "count-api-sidecar-proxy")
+	assertServiceOk(t, cc, "count-dashboard-sidecar-proxy")
```

Comment:
are service healthchecks adequate for confirming all the things we care about with this system?

 * services are up individually
 * one can talk to the other
 * they can't also bypass the mesh?
 * ... other?

Reply:
That's a good point... I should check on that dashboard health check to verify that it checks reachability. Bypass is a little harder to verify, because these applications are anemic container images. If reachability works that implies we've called `consul-cni` correctly, because it includes the DNS config. Most of the actual iptables rules details are owned by the plugin. I can see if there's improvements to be made there though.

Reply:
Well unfortunately the dashboard health doesn't include that, and the dashboard contents themselves are rendered via Javascript. So we can't even hit the dashboard alloc logs to find the "Fetched count" values:

```
$ nomad alloc logs -task dashboard 5a30
Starting server on http://0.0.0.0:9002
(Pass as PORT environment variable)
Using counting service at http://127.0.0.1:8080
(Pass as COUNTING_SERVICE_URL environment variable)
Starting websocket server...
New client connected
Fetched count 1
Fetched count 2
Fetched count 3
Fetched count 4
```

Because there's nothing to drive the Javascript! If we're going to continue to use countdash as our test application, adding an upstream reachability health check seems to be the way to go. Maybe there's some better choice we could use here, where the front server is proxying directly. I'll look into that.

Reply:
how about `exec`ing from within the task(s)? to exercise network from dashboard to api, from either to world, etc.

Reply:
I was going to say the container images are very bare-bones so there's nothing to exercise the network with, but then I remembered they have `busybox` and that has `wget`. Lemme give that a try.

Reply:
Assertion for upstream reachability added